### PR TITLE
feat(metrics): add info nox config metrics

### PR DESCRIPTION
**Pull request by kmd-fl**
## Description
Add Info metrics with the used nox config

## Proposed Changes
Used Nox Settings:
- Network Info
```
    pub tcp_port: u16,
    pub websocket_port: u16,
    pub listen_ip: String,
    pub network_type: String,
    pub bootstrap_nodes: Addresses,
    pub external_address: Option<String>,
    pub external_multiaddresses: Addresses,
```
- Vm Info
```
    pub allow_gpu: u8,
    pub public_ip: String,
    pub host_ssh_port: u16,
    pub vm_ssh_port: u16,
    pub port_range: String,
```
- System Info
```
    pub cpus_range: String,
    pub system_cpu_count: usize,
    pub particle_execution_timeout_sec: u64,
    pub max_spell_particle_ttl_sec: u64,
```

Looks like this:
```
# HELP nox_build Nox Info.
# TYPE nox_build info
nox_build_info{node_version="0.1.0",air_version="0.1.0",spell_version="0.1.0"} 1
# HELP nox_chain Chain Nox Info.
# TYPE nox_chain info
nox_chain_info{peer_id="peer_id",http_endpoint="",diamond_contract_address="",network_id="0",default_base_fee="",default_priority_fee="",ws_endpoint="",proof_poll_period_secs="0",min_batch_count="0",max_batch_count="0",max_proof_batch_size="0",epoch_end_window_secs="0"} 1
# HELP nox_network Network Nox Info.
# TYPE nox_network info
nox_network_info{tcp_port="0",websocket_port="0",listen_ip="",network_type="",bootstrap_nodes="",external_address="",external_multiaddresses=""} 1
# HELP nox_vm VM Nox Info.
# TYPE nox_vm info
nox_vm_info{allow_gpu="0",public_ip="",host_ssh_port="0",vm_ssh_port="0",port_range=""} 1
# HELP nox_system System Nox Info.
# TYPE nox_system info
nox_system_info{cpus_range="",system_cpu_count="0",particle_execution_timeout_sec="0",max_spell_particle_ttl_sec="0"} 1
# EOF
```

### CODE REVIEW

File: crates/server-config/src/node_config.rs [hunk 1]
```diff
@@ -212,6 +213,17 @@ impl TryFrom<&Network> for StreamProtocol {
     }
 }
 
+impl Display for Network {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Network::Dar => write!(f, "dar"),
```
Comment (folex) [inline crates/server-config/src/node_config.rs:219]: testnet

File: crates/server-config/src/node_config.rs [hunk 2]
```diff
@@ -212,6 +213,17 @@ impl TryFrom<&Network> for StreamProtocol {
...
+            Network::Stage => write!(f, "stage"),
+            Network::Kras => write!(f, "kras"),
```
Comment (folex) [inline crates/server-config/src/node_config.rs:221]: mainnet

**Review by folex — APPROVED** [review]
Awesome, I think this will be tremendously useful to observe our Providers!

File: crates/server-config/src/node_config.rs (hunk 1 above)
Comment (kmd-fl) [reply crates/server-config/src/node_config.rs:219]: Should we also change the names in the original Network enum? Will it break something?

File: crates/server-config/src/node_config.rs [hunk 3]
```diff
@@ -682,3 +694,9 @@ fn default_port_range_config() -> PortRangeConfig {
         end: 65535,
     }
 }
+
+impl Display for PortRangeConfig {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}-{}", self.start, self.end)
```
[the remark this answers is not in the document]
Comment (kmd-fl) [reply crates/server-config/src/node_config.rs:700]: Sure

File: crates/server-config/src/node_config.rs (hunk 1 above)
Comment (folex) [reply crates/server-config/src/node_config.rs:219]: We should.

Comment (kmd-fl) [reply crates/server-config/src/node_config.rs:219]: Will do it in a separate PR.

**Pull request merged**